#!/usr/bin/env python
PACKAGE = "dynamic_tutorials"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#units?
gen.add("speed_lim_v", double_t, 0, "Maximum linear velocity", 1.0, 0.0, 5.0)
gen.add("speed_lim_w", double_t, 0, "Maximum angular velocity", 2.0, 0.0, 10.0)

gen.add("velocity_factor", double_t, 0, "Linear constant",    1, -1,   1)
gen.add("deadlock_zone", double_t, 0, "DeadZone size",    .1, 0, .5)

gen.add("str_param",    str_t,    0, "Source frame",  "Hello World")
gen.add("str_param",    str_t,    0, "Target frame",  "Hello World")


exit(gen.generate(PACKAGE, "meka_guiding", "Guiding"))